cluster-autoscaler/core/scaledown/planner/planner.go

```diff
@@ -175,9 +177,33 @@ func (p *Planner) injectOngoingActuation() error {
 	if err != nil {
 		return err
 	}
-	// TODO(x13n): Check owner references to avoid double-counting already
-	// recreated pods.
-	return p.injectPods(p.actuationStatus.RecentEvictions())
+	return p.injectPods(filterOutRecreatedPods(p.actuationStatus.RecentEvictions(), p.cc))
+}
+
+func filterOutRecreatedPods(pods []*apiv1.Pod, cc ControllerReplicasCalculator) []*apiv1.Pod {
+	var podsToInject []*apiv1.Pod
+	addedReplicas := make(map[string]int32)
+	for _, pod := range pods {
+		if len(pod.GetOwnerReferences()) == 0 {
+			podsToInject = append(podsToInject, pod)
+			continue
+		}
+		ownerRef := pod.GetOwnerReferences()[0]
```

Comment:
I wonder if picking the first owner reference is always right. What if the first reference points to some custom resource, but the second one is a regular controller?

Reply:
I was under impression that pod will always have one owner, though just in case I added a function `getSupportedOwnerRef`